### PR TITLE
fix: Add Address model unmarshal func

### LINKED ISSUE #554: [V2] Add Address model unmarshal func
Since we define the Address interface to reusing the data struct but JSON unmarshal fun can't unmarshal []bye to the interface type.  
```
type Address interface {
	GetBaseAddress() BaseAddress
}

type IntervalAction struct {
	Timestamps
	Id           string
	Name         string
	IntervalName string
	Address      Address
}
```
This will cause error when unmarshaling from the DB binary data to model. So we need to custom the unmarshal function for IntervalAction and Subscription model.

### PR DESCRIPTION
**Pull request by weichou1229**
Close #554

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the current behavior?

JSON unmarshal fun can't unmarshal the interface type.

## What is the new behavior?
Add custom function for IntervalAction and Subscription model to unmarshal the Address.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/models/address.go [hunk 1]
```diff
@@ -5,10 +5,58 @@
...
+// FormatAddress convert the interface to the corresponding address type
+func FormatAddress(i interface{}) (address Address, err error) {
+	a, err := json.Marshal(i)
+	if err != nil {
+		return address, errors.NewCommonEdgeX(errors.KindContractInvalid, "Failed to marshal address.", err)
+	}
+	return UnmarshalAddress(a)
+}
+
+func UnmarshalAddress(b []byte) (address Address, err error) {
```
Comment (cloudxxx8) [inline v2/models/address.go:28]: use lower case for the func name

File: v2/models/address.go [hunk 2]
```diff
@@ -5,10 +5,58 @@
...
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+)
+
 type Address interface {
 	GetBaseAddress() BaseAddress
 }
 
+// FormatAddress convert the interface to the corresponding address type
+func FormatAddress(i interface{}) (address Address, err error) {
```
Comment (cloudxxx8) [inline v2/models/address.go:20]: use lower case for the func name
rename to `instantiateAddress`

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v2/models/address.go (hunk 1 above)
Comment (weichou1229) [reply v2/models/address.go]: Modified.

File: v2/models/address.go (hunk 2 above)
Comment (weichou1229) [reply v2/models/address.go]: Renamed.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**